Doc/c-api/mapping.rst

```diff
@@ -8,8 +8,11 @@ Mapping Protocol
 
 .. c:function:: int PyMapping_Check(PyObject *o)
 
-   Return ``1`` if the object provides mapping protocol, and ``0`` otherwise.  This
-   function always succeeds.
+   **Not recommended.** Return ``1`` if the object provides the C-API mapping
+   protocol, and ``0`` otherwise.  This function always succeeds.  The C-API
```

Comment:
I recommend using @serhiy-storchaka's suggested wording to move this PR forward.

~~The C-API mapping protocol is not equivalent to :class:`collections.abc.Mapping`, and also returns ``1`` for sequences that support slicing.  Use :c:func::`PyObject_IsInstance` instead.~~

`PyMapping_Check() also returns 1 for sequences that support slicing.`